src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1454,15 +1457,18 @@ def _update_container_app_source(self, cmd, registry_server, registry_user, regi
 
         # Set image to None if it was previously set to the default image (case where image was not provided by the user) else reformat it
         image = None if self.get_argument_image() is None else _reformat_image(source=self.get_argument_source(), image=self.get_argument_image(), repo=None)
-        location = self.containerapp_def["location"]
+
+         # Parse location
+        env_info = self.get_environment_client().show(cmd=self.cmd, resource_group_name=env_rg, name=env_name)
+        location = self.containerapp_def["location"] if "location" in self.containerapp_def else env_info['location']
 
         has_dockerfile = _has_dockerfile(self.get_argument_source(), dockerfile)
         if has_dockerfile:
             dockerfile_content = _get_dockerfile_content(repo=None, branch=None, token=None, source=self.get_argument_source(), context_path=None, dockerfile=dockerfile)
             ingress, target_port = _get_ingress_and_target_port(self.get_argument_ingress(), self.get_argument_target_port(), dockerfile_content)
 
         # Construct ContainerApp
-        resource_group = ResourceGroup(cmd, env_rg, location=location)
+        resource_group = ResourceGroup(cmd, self.get_argument_resource_group_name(), location=location)
         env = ContainerAppEnvironment(cmd, env_name, resource_group, location=location)
```

Comment:
Same as above, please use 2 different values for env's and app's `resource_group `.

Reply:
Same comment as above.